src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -148,38 +177,55 @@ private void CheckDisposed()
             }
         }
 
-        private void WriteInternal(byte[] buffer, int offset, int count)
+        private async Task InternalWriteAsync(byte[] buffer, int offset, int count, CancellationToken token)
         {
             if (_chunkedMode)
             {
```

Comment:
Isn't WinHttp handling chunked internally? This looks like it's implementing the chunked protocol.

https://msdn.microsoft.com/en-us/library/windows/desktop/aa384110(v=vs.85).aspx:
> Starting in Windows Vista and Windows Server 2008, WinHttp enables applications to perform chunked transfer encoding on data sent to the server.